From 2c9da4c8c64685f229fd35f17e871999eac9879f Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 5 Mar 2019 09:09:48 +0100
Subject: [PATCH 2/2] [DNM] Squash of lineage-16.0-android-9.0.0_r34

Squashed commit of the following:

commit d6afaec61a6a4114d0f89f72d2b7404376061595
Merge: 1a7ca2f 35f34c2
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Jan 16 18:58:14 2019 +0000

    Merge cherrypicks of [6072697, 6072075, 6072758, 6072124, 6072885, 6072886, 6072887, 6072580, 6072581, 6072582, 6072583, 6072584, 6072132, 6072195, 6072133, 6072077, 6072134, 6072078, 6072211, 6072762, 6072763, 6072908, 6072909, 6072910, 6072911, 6072912, 6072913, 6072914, 6072930, 6072212, 6072743] into pi-qpr2-release

    Change-Id: I60ab03dbcb39de7c6f7e959ff063088868e8898e

commit 35f34c2a18a09660e63cd509a2c37eee70893b67
Author: George Chang <georgekgchang@google.com>
Date:   Wed Nov 7 22:44:56 2018 +0800

    Prevent Out of bound error in phNxpNciHal_process_ext_rsp

    Bug: 118152591
    Test: Nfc Enable/Disable, R/W, P2P
    Merged-In: I53bfc1b7eca4c3306f20488dc5fb8ccf9ed0e330
    Change-Id: I53bfc1b7eca4c3306f20488dc5fb8ccf9ed0e330
    (cherry picked from commit 210180d4eb8971f74aa17d1677e97a342c29c7b1)

Change-Id: I7187a1379dbaceb505fb7ca9af84128a300d1167
---
 halimpl/hal/phNxpNciHal_ext.cc | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/halimpl/hal/phNxpNciHal_ext.cc b/halimpl/hal/phNxpNciHal_ext.cc
index 3dea0e7..372a147 100755
--- a/halimpl/hal/phNxpNciHal_ext.cc
+++ b/halimpl/hal/phNxpNciHal_ext.cc
@@ -12,8 +12,8 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- */#include <log/log.h>
-
+ */
+#include <log/log.h>
 #include <phDal4Nfc_messageQueueLib.h>
 #include <phNxpConfig.h>
 #include <phNxpLog.h>
@@ -100,6 +100,15 @@ NFCSTATUS phNxpNciHal_process_ext_rsp(uint8_t* p_ntf, uint16_t* p_len) {
   NFCSTATUS status = NFCSTATUS_SUCCESS;
   uint16_t rf_technology_length_param = 0;
 
+  if (p_ntf[0] == 0x61 && p_ntf[1] == 0x05 && *p_len < 14) {
+    if(*p_len <= 6) {
+      android_errorWriteLog(0x534e4554, "118152591");
+    }
+    NXPLOG_NCIHAL_E("RF_INTF_ACTIVATED_NTF length error!");
+    status = NFCSTATUS_FAILED;
+    return status;
+  }
+
   if (p_ntf[0] == 0x61 && p_ntf[1] == 0x05 && p_ntf[4] == 0x03 &&
       p_ntf[5] == 0x05 && nxpprofile_ctrl.profile_type == EMV_CO_PROFILE) {
     p_ntf[4] = 0xFF;
-- 
2.17.1

